### DISCUSSION TITLE
Crafting table

### DISCUSSION THREAD

**Discussion by opekope2** [opekope2/StaffMod#13] [Staff blocks]
Crafting table is shown at [00:45-00:48](https://youtu.be/vlVUVsx6BYY?t=45). It picks ups items from the ground, and spits out items, blocks, and entities.

| Item            | Ingredients                                                                  |
|-----------------|------------------------------------------------------------------------------|
| Shield          | 6 planks, 1 iron                                                             |
| Bed             | 3 planks, 3 wool                                                             |
| Sign            | 1 stick, 6 planks                                                            |
| Diamond pickaxe | 3 diamonds, 2 sticks                                                         |
| Grindstone      | 2 planks, 2 sticks, 1 smooth stone slab                                      |
| Boat            | 5 planks                                                                     |
| Torch           | 1 coal, 1 stick                                                              |
| Chest           | 8 planks                                                                     |
|                 |                                                                              |
| Total           | 30 planks, 1 iron, 3 wool, 6 sticks, 3 diamonds, 1 smooth stone slab, 1 coal |
| Video           | 5 planks, 3 iron, 3 wool, 5 sticks, 3 diamonds, 1 smooth stone slab, 1 coal  |

Implementation idea: press and hold the `use` key (right click) to collect items, release it to craft them (randomly? start with the recipe with most ingredients?)
Attack damage: same as empty staff (?)
Equip time: same as empty staff (?)

**Comment by opekope2** [comment]
Another implementation idea is to specify preferred recipes/items, and only craft those/craft those first, then the others

**Reply by charmander9695** [reply]
yeah you can do it like that, that would be a great idea

**Comment by charmander9695** [comment]
I think there are two ways to put it but if you find more then there you go
1 open the crafting gui
2have a way to take items, in or out into a line of crafts you click on

**Comment by opekope2** [comment]
I just had another idea: 
When releasing right click, the crafting recipe book shows up, then the player can click on items to craft them. When the player closes the screen, all non-crafted items will be ejected.
